### PR TITLE
Rewritten F18 export and optimized exporter

### PR DESCRIPTION
**Pull request by syn111**
I rewrote the F18 exporter to fix some of the current glitches, namely:
- When selecting MDIS A/B channels and exiting via CLR, without selecting a different option, the preset is not saved and the MIDS radio turns off in SRS
- When introducing a new IFF code, the code gets registered with SRS without pressing ENT in DCS, and thus SRS squawks a different code than what DCS has in the UFC
- The exporter crash issue as discussed in issue #542 and for what I had already proposed a fix on #548 
- Also fixed IFF not defaulting to on on hot start (code was in place on your file but didn't work)

The main reason for the 'desync' glitches was the fact that the previous code couldn't intercept the ENT key being pressed in DCS. I have fixed that by intercepting the ENT keypress, and reading the value from the UFC after ENT is pressed, both for MIDS and for the transponder modes. The IFF modes are also read on each iteration to parse any mode selection changes.

Adding the code for intercepting the ENT keypress required me modifying LuaExportBeforeNextFrame (to make sure it gets captured) and LuaExportActivityNextEvent (to save the last aircraft type), and thus I also noticed a few inefficiencies taking into account the internal workings of LUA. As the fix already required modifying those two functions, I also made some streamlining in those two functions, namely rewriting all the entangled "if _update.unit == "Mi-8MT"'s into a table lookup (_exporters), and by modifying the pcall calls so that there is no dynamic creation of functions happening every tenth of a second. 

Additionally, I changed the "_send" logic that made LuaExportActivityNextEvent be called every 0.1 seconds ("for compatibility with Helios") to a logic that calls LuaExportActivityNextEvent every 0.2 seconds unless another call in the chain requires a lower update rate. In that cae, LuaExportActivityNextEvent returns the tNext required by that function, but the SRS code is only executed every 0.2 seconds regardless.

TLDR: the Hornet's UFC interface now should be more solid and without any glitches, and the SRS exporter takes less cpu time to execute now.

I submit this pull request, looking to help improve SRS, and I'm happy to discuss any issues you find with it.

### CODE REVIEW

**Comment by ciribob** [comment]
Thanks for this! 

For the exporters lookup at the bottom - is that any faster / more efficient than the string comparison / lookup? 

I'm not sure of the inner workings of a table lookup in LUA so genuinely curious if its worth doing vs the string lookup.

My only (weak) argument for keeping the String method - is when ED release variants of aircrafts, like the P47, I dont have to rush out an update. Its a minor issue though but genuinely curious around the performance benefits 

It is cleaner code wise for sure -  which is probably reason enough to switch either way

**Comment by syn111** [comment]
Hi cribob,

For non-numeric values, LUA tables work as hash tables and, as the lookup is performed on the "C side", there is a performance gain there too. I ran a quick benchmark test and the lookup method was 16 times faster than the "entangled" method for a key on the middle zone.

Of course, as the code in this case is executed only 5 times per second, the performance gain is negligible, and you can have a point with your argument about the current method and ED's new module releases.

**Comment by ciribob** [comment]
I'll be sorting a release this weekend with this in. Thanks again!
